numba/unicode.py

```diff
@@ -594,6 +594,32 @@ def split_whitespace_impl(a, sep=None, maxsplit=-1):
         return split_whitespace_impl
 
 
+@overload_method(types.UnicodeType, 'center')
+def unicode_center(string, width, fillchar=' '):
+    if not isinstance(width, types.Integer):
+        raise TypingError('The width must be an Integer')
+    if not (fillchar == ' ' or isinstance(fillchar, (types.Omitted, types.UnicodeType))):
+        raise TypingError('The fillchar must be a UnicodeType')
+
+    def center_impl(string, width, fillchar=' '):
+        str_len = len(string)
+        fillchar_len = len(fillchar)
+
+        if fillchar_len != 1:
+            raise ValueError('The fill character must be exactly one character long')
+
+        if width <= str_len:
+            return string
+
+        allmargin = width - str_len
+        lmarging = (allmargin // 2) + (allmargin & width & 1)
```

Comment:
```suggestion
        lmargin = (allmargin // 2) + (allmargin & width & 1)
```
(and then fix up throughout)